src/components/ReportActionItem/MoneyRequestPreview/index.tsx

```diff
@@ -0,0 +1,42 @@
+// We should not render the component if there is no iouReport and it's not a split.
+// Moved outside of the component scope to allow memoization of values later.
+import lodashIsEmpty from 'lodash/isEmpty';
+import React from 'react';
+import {withOnyx} from 'react-native-onyx';
+import * as ReportActionsUtils from '@libs/ReportActionsUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import MoneyRequestPreviewContent from './MoneyRequestPreviewContent';
+import type {MoneyRequestPreviewOnyxProps, MoneyRequestPreviewProps} from './types';
+
+// We should not render the component if there is no iouReport and it's not a split.
+// Moved outside of the component scope to allow for easier use of hooks in the main component.
+function MoneyRequestPreview(props: MoneyRequestPreviewProps) {
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    return lodashIsEmpty(props.iouReport) && !props.isBillSplit ? null : <MoneyRequestPreviewContent {...props} />;
+}
+
+MoneyRequestPreview.displayName = 'MoneyRequestPreview';
+
+export default withOnyx<MoneyRequestPreviewProps, MoneyRequestPreviewOnyxProps>({
+    personalDetails: {
+        key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+    },
+    chatReport: {
+        key: ({chatReportID}) => `${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`,
+    },
+    iouReport: {
+        key: ({iouReportID}) => `${ONYXKEYS.COLLECTION.REPORT}${iouReportID}`,
+    },
+    session: {
+        key: ONYXKEYS.SESSION,
+    },
+    transaction: {
+        key: ({action}) => `${ONYXKEYS.COLLECTION.TRANSACTION}${ReportActionsUtils.isMoneyRequestAction(action) ? action?.originalMessage?.IOUTransactionID : 0 ?? 0}`,
```

Comment:
The previous version of this was more readable, can we break this into a couple of lines?

Reply:
done